bindings/matrix-sdk-ffi/src/timeline/mod.rs

```diff
@@ -600,16 +608,26 @@ impl Timeline {
     /// local events that are being processed.
     pub async fn redact_event(
         &self,
-        item: Arc<EventTimelineItem>,
+        id: String,
         reason: Option<String>,
     ) -> Result<bool, ClientError> {
-        let removed = self
-            .inner
-            .redact(&item.0, reason.as_deref())
-            .await
-            .map_err(|err| anyhow::anyhow!(err))?;
+        let event = if let Ok(event_id) = EventId::parse(&id) {
+            self.inner.item_by_event_id(&event_id).await
+        } else {
+            let transaction_id: OwnedTransactionId = id.into();
+            self.inner.local_item_by_transaction_id(&transaction_id).await
+        };
+        if let Some(event) = event {
+            let removed = self
+                .inner
+                .redact(&event, reason.as_deref())
+                .await
+                .map_err(|err| anyhow::anyhow!(err))?;
 
-        Ok(removed)
+            Ok(removed)
+        } else {
+            Ok(false)
+        }
```

Comment:
Why the changes here? What's an "id"? Is it an event id? a transaction id? a timeline item id?

Reply:
> Why the changes here?

The previous `ffi::EventTimelineItem` implementation used to have a `ui::EventTimelineItem` inside it which we could use for stuff like cloning the contents, getting a send handle, etc. But that's no longer the case since `EventTimelineItem` is now a record type. What we use instead is an id.

> What's an "id"? Is it an event id? a transaction id?

In this case, it can be any of those two (and I need to add docs for that). Since we don't really have a wrapper for the event/transaction/unique ids in the FFI layer (for performance reasons?), I can think of several options when we can to be able to pass both event ids and transaction ids:

- Use this strategy, parse the id once received, I chose this one..
- Create those wrappers, although I don't know how that could affect both performance and the existing APIs.
- Create separate `redact_event_by_event_id`, `redact_event_by_transaction_id` fns.
- Use the `UniqueId`: I already tried this in the past, but it has some issues with the current APIs.

Reply:
Or name the argument `event_or_transaction_id`?

Reply:
That I can do!